mobile-academy/src/main/java/org/motechproject/nms/mobileacademy/service/impl/SmsNotificationServiceImpl.java

```diff
@@ -0,0 +1,162 @@
+package org.motechproject.nms.mobileacademy.service.impl;
+
+import org.apache.commons.httpclient.HttpStatus;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.util.EntityUtils;
+import org.joda.time.DateTime;
+import org.motechproject.event.MotechEvent;
+import org.motechproject.event.listener.annotations.MotechListener;
+import org.motechproject.nms.mobileacademy.domain.CompletionRecord;
+import org.motechproject.nms.mobileacademy.repository.CompletionRecordDataService;
+import org.motechproject.nms.mobileacademy.service.SmsNotificationService;
+import org.motechproject.server.config.SettingsFacade;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+import java.io.UnsupportedEncodingException;
+
+/**
+ * This handles all the integration pieces between MA and sms module to trigger and handle notifications
+ * for course completion
+ */
+@Service("smsNotificationService")
+public class SmsNotificationServiceImpl implements SmsNotificationService {
+
+    private static final String COURSE_COMPLETED = "nms.ma.course.completed";
+
+    private static final String SMS_STATUS = "nms.ma.sms.deliveryStatus";
+
+    private static final String SMS_NOTIFICATION_URL = "imi.sms.notification.url";
+
+    private static final String SMS_MESSAGE_CONTENT = "imi.sms.course.completion.message";
+
+    private static final String CALLBACK_URL = "imi.sms.status.callback.url";
+
+    private static final String SMS_SENDER_ID = "imi.sms.sender.id";
+
+    private static final String SMS_TEMPLATE_FILE = "smsTemplate.json";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SmsNotificationServiceImpl.class);
+
+    private CompletionRecordDataService completionRecordDataService;
+
+    private SettingsFacade settingsFacade;
+
+    @Autowired
+    public SmsNotificationServiceImpl(CompletionRecordDataService completionRecordDataService,
+                                      @Qualifier("maImiSettings") SettingsFacade settingsFacade) {
+        this.completionRecordDataService = completionRecordDataService;
+        this.settingsFacade = settingsFacade;
+    }
+
+    @MotechListener(subjects = { COURSE_COMPLETED })
+    public void sendSmsNotification(MotechEvent event) {
+
+        LOGGER.debug("Handling course completion notification event");
```

Comment:
Consistency is the hobgoblin of foolish minds.  Why is the first line in this handler blank followed by the logger while in the next handler it is reversed? :trollface: